rust/src/io/exec/projection.rs

```diff
@@ -89,7 +89,28 @@ impl Stream for ProjectionStream {
     type Item = DataFusionResult<RecordBatch>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        Pin::into_inner(self).rx.poll_recv(cx)
+        let this = Pin::into_inner(self);
+        // We need to check the JoinHandle to make sure the thread hasn't panicked.
+        let bg_thread_completed = if let Some(bg_thread) = &mut this.bg_thread {
```

Comment:
This is why we missed the performance hit in our original benchmark: the filter during deletion was failing silently because we weren't picking up the panic in the background thread. Since filter was failing, the rows we never being deleted.